query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java

```diff
@@ -486,7 +486,7 @@ private String convertLiteralToString(LiteralConstant literal, Builder paramsBui
         break;
       case BOOL:
         ret = QUESTION_MARK;
-        paramsBuilder.addStringParam(String.valueOf(value.getBoolean()));
+        paramsBuilder.addBooleanParam(value.getBoolean());
```

Comment:
note: Recently, we added the support for BOOLEAN_ARRAY too. So, when we merge it, we need to apply boolean type there too.